src/main/java/org/example/Promotion.java

```diff
@@ -0,0 +1,128 @@
+        Department kikaku = new Department(2, "企画部");
+        Department eigyo = new Department(3, "営業部");
+        Department gijutu = new Department(4, "技術部");
+
+        // 資格クラスのインスタンス化
+        Qualification level1 = new Qualification(1, "資格レベル１", 1);
+        Qualification level2 = new Qualification(2, "資格レベル２", 5);
+        Qualification level3 = new Qualification(3, "資格レベル３", 8);
+
+        // 従業員リストの作成、従業員クラスのインスタンス化
+        List<Employee> employeeList = new ArrayList<>();
+
+        Employee soumuA = new Employee(1, "soumuA", 12, soumu, level3);
+        Employee soumuB = new Employee(2, "soumuB", 8, soumu, level1);
+        Employee soumuC = new Employee(3, "soumuC", 6, soumu, level2);
+        Employee kikakuA = new Employee(4, "kikakuA", 8, kikaku, level1);
+        Employee kikakuB = new Employee(5, "kikakuB", 7, kikaku, level2);
+        Employee kikakuC = new Employee(6, "kikakuC", 1, kikaku, level2);
+        Employee eigyoA = new Employee(7, "eigyoA", 10, eigyo, level3);
+        Employee eigyoB = new Employee(8, "eigyoB", 8, eigyo, level2);
+        Employee eigyoC = new Employee(9, "eigyoC", 3, eigyo, level1);
+        Employee gijutuA = new Employee(10, "gijutuA", 15, gijutu, level2);
+        Employee gijutuB = new Employee(11, "gijutuB", 12, gijutu, level2);
+        Employee gijutuC = new Employee(12, "gijutuC", 12, gijutu, level3);
+        Employee gijutuD = new Employee(13, "gijutuD", 15, gijutu, level2);
+        Employee gijutuE = new Employee(14, "gijutuE", 8, gijutu, level3);
+
+        employeeList.add(soumuA);
+        employeeList.add(soumuB);
+        employeeList.add(soumuC);
+        employeeList.add(kikakuA);
+        employeeList.add(kikakuB);
+        employeeList.add(kikakuC);
+        employeeList.add(eigyoA);
+        employeeList.add(eigyoB);
+        employeeList.add(eigyoC);
+        employeeList.add(gijutuA);
+        employeeList.add(gijutuB);
+        employeeList.add(gijutuC);
+        employeeList.add(gijutuD);
+        employeeList.add(gijutuE);
+
+        //Scannerクラスを使ってユーザーに部署コードを入力させ、入力結果に応じた出力を行う。
+        Scanner scanner = new Scanner(System.in);
+        while (true) {
+            try {
+                System.out.println("総務部：1、企画部：2、営業部：3、技術部：4");
+                System.out.print("昇進者を抽出したい部署のコード（1~4）を入力してください：");
+
+                //整数が入力されていない場合は即座にエラーを返す。整数が入力された場合は無視される。
+                if (!scanner.hasNextInt()) {
+                    throw new IllegalArgumentException("部署コード以外は入力できません!");
+                }
+
+                // 入力された整数をinputCodeに代入する。
+                int inputCode = scanner.nextInt();
+                scanner.nextLine(); //入力後の改行をクリア
+
+                // inputCodeに1~4以外が入力された場合は、エラーを返し、1~4が入力された場合はその部署の昇進者を表示する。
+                if (inputCode < 1 || inputCode > 4) {
+                    throw new IllegalArgumentException("部署コード以外は入力できません!");
+                } else {
+                    List<Employee> promotionList = employeeList.stream()
```

Comment:
昇進する社員なので
```suggestion
                    List<Employee> promotingEmployees = employeeList.stream()
```
が良いかなと思いました。

Reply:
ご指摘ありがとうございます！
対応しました！[8535f3a](https://github.com/YaraiM/JavaChapter4Assignment/pull/1/commits/8535f3a191e3d5593aae5b079f6b2977bbfa6cef)